core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -365,55 +365,65 @@ class WorkspaceService(protected val userInfo: UserInfo,
-            _ <- dataAccess.methodConfigurationQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
-            _ <- dataAccess.entityQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+  private def deleteWorkspaceTransaction(workspaceName: WorkspaceName, workspaceContext: Workspace) = {
+    dataSource.inTransaction { dataAccess =>
+      for {
+        // Delete components of the workspace
+        _ <- dataAccess.submissionQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+        _ <- dataAccess.methodConfigurationQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
+        _ <- dataAccess.entityQuery.deleteFromDb(workspaceContext.workspaceIdAsUUID)
 
-            // Delete the workspace
-            _ <- dataAccess.workspaceQuery.delete(workspaceName)
+        // Schedule bucket for deletion
+        _ <- dataAccess.pendingBucketDeletionQuery.save(PendingBucketDeletionRecord(workspaceContext.bucketName))
 
-            // Schedule bucket for deletion
-            _ <- dataAccess.pendingBucketDeletionQuery.save(PendingBucketDeletionRecord(workspaceContext.bucketName))
+        // Delete the workspace
+        _ <- dataAccess.workspaceQuery.delete(workspaceName)
+      } yield ()
+    }
+  }
 
-          } yield {
-            workflowsToAbort
-          }
+  private def deleteWorkspace(workspaceName: WorkspaceName, workspaceContext: Workspace): Future[PerRequestMessage] = {
+    for {
+      _ <- requesterPaysSetupService.revokeAllUsersFromWorkspace(workspaceContext) recoverWith {
+        case t:Throwable => {
+          logger.warn(s"Unexpected failure deleting workspace (while revoking 'requester pays' users) for workspace `${workspaceName}`", t)
+          throw t
         }
       }
-    for {
 
-      workflowsToAbort <- deletionFuture recoverWith {
+      workflowsToAbort <- gatherWorkflowsToAbortAndSetStatusToAborted(workspaceName, workspaceContext) recoverWith {
         case t:Throwable => {
-          logger.warn(s"Unexpected failure deleting workspace (while running `deletionFuture`) for workspace `${workspaceName}`", t)
+          logger.warn(s"Unexpected failure deleting workspace (while gathering workflows that need to be aborted) for workspace `${workspaceName}`", t)
           throw t
         }
       }
 
-      // Abort running workflows
+      //Attempt to abort any running workflows so they don't write any more to the bucket.
+      //Notice that we're kicking off Futures to do the aborts concurrently, but we never collect their results!
+      //This is because there's nothing we can do if Cromwell fails, so we might as well move on and let the
+      //ExecutionContext run the futures whenever
       aborts = Future.traverse(workflowsToAbort) { wf => executionServiceCluster.abort(wf, userInfo) } recoverWith {
         case t:Throwable => {
           logger.warn(s"Unexpected failure deleting workspace (while aborting workflows) for workspace `${workspaceName}`", t)
```

Comment:
Why bother even logging this? Since we're doing this asynchronously and don't care about the outcome, why log it?

Reply:
This probably is unnecessary but it was added in to try to log any issues in deleteWorkspace, again to help with debugging the root cause of CA-1206